### PR TITLE
feat: added utils/every-in-by

### LINKED ISSUE #822: [RFC]: Add `@stdlib/utils/every-in-by`
### Description

This RFC proposes adding a utility to test whether every "own" and inherited property of a provided object satisfies a predicate function.

This is the `for-in` equivalent of [`@stdlib/utils/every-by`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/utils/every-by).

Package: `@stdlib/utils/every-in-by`
Alias: `everyInBy`

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by saisrikardumpeti**
Resolves #822 

## Description

> What is the purpose of this pull request?

This pull request:

-   a
-   b
-   c

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Planeshifter** [comment]
@the-r3aper7 This package is currently missing a `README.md` file. Would you mind adding it, please?

**Comment by saisrikardumpeti** [comment]
@Planeshifter i have added `README.md` file.

File: lib/node_modules/@stdlib/utils/every-in-by/README.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# every in by
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/README.md]:
```suggestion
# everyInBy
```

File: lib/node_modules/@stdlib/utils/every-in-by/README.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+> Test whether all elements in a object are truthy.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/README.md]:
```suggestion
> Test whether all properties (own and inherited) of an object pass a test implemented by a predicate function.
```

File: lib/node_modules/@stdlib/utils/every-in-by/README.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+## Usage
+
+```javascript
+var everyInBy = require( '@stdlib/utils/every-in-by' );
+```
+
+#### everyInBy( object, predicate\[, thisArg ] )
+
+Tests whether all elements in a `object` pass a test implemented by a `predicate` function.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/README.md]:
```suggestion
Tests whether all properties (own and inherited) of an `object` pass a test implemented by a `predicate` function.
```

File: lib/node_modules/@stdlib/utils/every-in-by/README.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+* * *
+
+## See Also
+
+-   <span class="package-name">[`@stdlib/utils/any`][@stdlib/utils/any]</span><span class="delimiter">: </span><span class="description">test whether at least one element in a object is truthy.</span>
+-   <span class="package-name">[`@stdlib/utils/every-by`][@stdlib/utils/every-by]</span><span class="delimiter">: </span><span class="description">test whether all elements in a object pass a test implemented by a predicate function.</span>
+-   <span class="package-name">[`@stdlib/utils/for-each`][@stdlib/utils/for-each]</span><span class="delimiter">: </span><span class="description">invoke a function for each element in a object.</span>
+-   <span class="package-name">[`@stdlib/utils/none`][@stdlib/utils/none]</span><span class="delimiter">: </span><span class="description">test whether all elements in a object are falsy.</span>
+-   <span class="package-name">[`@stdlib/utils/some`][@stdlib/utils/some]</span><span class="delimiter">: </span><span class="description">test whether a object contains at least `n` elements which are truthy.</span>
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/README.md]: We auto-populate this section, no reason to fill in manually.

```suggestion
```

File: lib/node_modules/@stdlib/utils/every-in-by/README.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+[@stdlib/utils/any]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/utils/any
+
+[@stdlib/utils/every-by]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/utils/every-by
+
+[@stdlib/utils/for-each]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/utils/for-each
+
+[@stdlib/utils/none]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/utils/none
+
+[@stdlib/utils/some]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/utils/some
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/utils/every-in-by/lib/index.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+/**
+* Test whether all elements in a collection pass a test implemented by a predicate function, iterating from left to right.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/lib/index.js]:
```suggestion
* Test whether all elements in an object pass a test implemented by a predicate function.
```

File: lib/node_modules/@stdlib/utils/every-in-by/lib/main.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+var isObject = require( '@stdlib/assert/is-object' );
+var isFunction = require( '@stdlib/assert/is-function' );
+var format = require( '@stdlib/string/format' );
+
+
+// MAIN //
+
+/**
+* Tests whether all elements in a object pass a test implemented by a predicate function.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/lib/main.js]:
```suggestion
* Tests whether all properties (own and inherited) of an object pass a test implemented by a predicate function.
```

File: lib/node_modules/@stdlib/utils/every-in-by/lib/main.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+*
+* @param {Object} obj - input object
+* @param {Function} predicate - test function
+* @param {*} [thisArg] - execution context
+* @throws {TypeError} first argument must be a object
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/lib/main.js]:
```suggestion
* @throws {TypeError} first argument must be an object
```

File: lib/node_modules/@stdlib/utils/every-in-by/package.json [hunk 9]
```diff
@@ -0,0 +1,70 @@
+{
+  "name": "@stdlib/utils/every-in-by",
+  "version": "0.0.0",
+  "description": "Test whether all elements in a object pass a test implemented by a predicate function, iterating from left to right.",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/package.json]:
```suggestion
  "description": "Test whether all properties (own and inherited) of an object pass a test implemented by a predicate function.",
```

File: lib/node_modules/@stdlib/utils/every-in-by/docs/repl.txt [hunk 10]
```diff
@@ -0,0 +1,43 @@
+
+{{alias}}( object, predicate[, thisArg ] )
+    Tests whether all elements in a object pass a test implemented by a
+    predicate function, iterating from right to left.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/docs/repl.txt]:
```suggestion
    Test whether all properties (own and inherited) of an object pass a 
    test implemented by a predicate function.
```

File: lib/node_modules/@stdlib/utils/every-in-by/package.json [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    "util",
+    "test",
+    "predicate",
+    "every",
+    "all",
+    "everyright",
+    "array.every",
+    "iterate",
+    "collection",
+    "array-like",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/package.json]:
```suggestion
    "object",
    "iterate",
```

File: lib/node_modules/@stdlib/utils/every-in-by/test/test.js [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require('tape');
+var noop = require('@stdlib/utils/noop');
+var everyInBy = require( './../lib' ); // Replace './lib' with the actual path to your function
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/test/test.js]:
```suggestion
var everyInBy = require( './../lib' );
```

File: lib/node_modules/@stdlib/utils/every-in-by/test/test.js [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require('tape');
+var noop = require('@stdlib/utils/noop');
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/test/test.js]: Should insert spaces here and below around parentheses to match project conventions:

```suggestion
var tape = require( 'tape' );
var noop = require( '@stdlib/utils/noop' );
```

File: lib/node_modules/@stdlib/utils/every-in-by/lib/main.js [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+	}
+	if ( !isFunction( predicate ) ) {
+		throw new TypeError( format( 'invalid argument. Second argument must be a function. Value: `%s`.', predicate ) );
+	}
+
+	for (key in obj) {
+		if (!predicate.call(thisArg, obj[key], key, obj)) {
+			return false;
+		}
+	}
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/lib/main.js]:
```suggestion
	for ( key in obj ) { // eslint-disable-line guard-for-in
		if ( !predicate.call( thisArg, obj[ key ], key, obj ) ) {
			return false;
		}
	}
```

File: lib/node_modules/@stdlib/utils/every-in-by/docs/types/index.d.ts [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// Define types for functions/arguments
+type Predicate<T> = ( value: T, key: string, obj: object ) => boolean;
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/docs/types/index.d.ts]: Please have a look at the type definitions for the predicate function in say `every-by`, which, even though for collections / arrays, should give some indication of how we want to type the `Predicate` function as well as `everyInBy`

**Review by Planeshifter — CHANGES_REQUESTED** [review]

**Comment by saisrikardumpeti** [comment]
@Planeshifter i have made the changes as you suggested.

File: lib/node_modules/@stdlib/utils/every-by/docs/types/index.d.ts [hunk 16]
```diff
@@ -20,7 +20,7 @@
 
 /// <reference types="@stdlib/types"/>
 
-import { Collection } from '@stdlib/types/array';
+import { Collection } from '@stdlib/types/object';
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-by/docs/types/index.d.ts]: Need to revert this change:

```suggestion
import { Collection } from '@stdlib/types/array';
```

File: lib/node_modules/@stdlib/utils/every-in-by/docs/repl.txt [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+
+    If provided an empty object, the function returns `true`.
+
+    Parameters
+    ----------
+    object: Object
+        Input object over which to iterate.
+
+    predicate: Function
+        The test function.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/docs/repl.txt]:
```suggestion
        Test function.
```

File: lib/node_modules/@stdlib/utils/every-in-by/docs/types/index.d.ts [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+ *
+ * @throws TypeError if `obj` is not an object or if `predicate` is not a function.
+ */
+declare function everyInBy<T, U>(
+	obj: object,
+	predicate: Predicate<T, U>,
+	thisArg?: ThisParameterType<Predicate<T, U>>
+): boolean;
+
+// Export the function
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/docs/types/index.d.ts]:
```suggestion

// EXPORTS //

```

File: lib/node_modules/@stdlib/utils/every-in-by/docs/types/index.d.ts [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+/**
+* Checks whether an element in a collection passes a test.
+*
+* @param value - collection value
+* @param index - collection index
+* @param collection - input collection
+* @returns boolean indicating whether an element in a collection passes a test
+*/
+type Predicate<T, U> = Nullary<U> | Unary<T, U> | Binary<T, U> | Ternary<T, U>;
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/every-in-by/docs/types/index.d.ts:64]:
```suggestion
```

**Review by Planeshifter — APPROVED** [review]
Made a few changes to get this over the finish line; will merge once CI passes.

Thanks for your contribution!

**Pull request merged**